### ISSUE TITLE
Drop down visibility causes DxComboBox to crash

### ISSUE THREAD

**Issue by emysa341** [DevExpress/Blazor#62]
I am using DxComboBox to display some options and If I use ListRenderMode then drop down list is not visible so I enabled it using @bind-DropDownVisible. But as soon as I use that I get exceptions. Following usage of DxComboBox works

`<DxComboBox Data="@(new string[] {"Option 1", "Option 2", "Option 3" })"   
            @bind-SelectedItem="SelectedStartStation"
            NullText="Choose option"
            AllowUserInput="true"
            FilteringMode="DataGridFilteringMode.Contains"     
            ListRenderMode="ListRenderMode.Virtual"
            ></DxComboBox>`
**- Following fails with errors:**
`<DxComboBox Data="@(new string[] {"Option 1", "Option 2", "Option 3" })"   
            @bind-SelectedItem="SelectedStartStation"
            NullText="Choose start station"
            AllowUserInput="true"
            FilteringMode="DataGridFilteringMode.Contains"     
            ListRenderMode="ListRenderMode.Virtual"
            @bind-DropDownVisible="@DropDownVisible"
            ></DxComboBox>`
![image](https://user-images.githubusercontent.com/5163679/67076760-1c106f80-f18e-11e9-8109-bfc7d9d1a5e5.png)

**IEnumerable<T> Datasource:**
In addition to that, if I use generic IEnumerable<MyCustomClass> data source and even specify the TextFieldName I get errors and as soon as change it to IEnumerable<string>, there are no errors. Following are the errors

`The type arguments for method 'TypeInference.CreateDxComboBox_0<T>(RenderTreeBuilder, int, int, IEnumerable<T>, int, string, int, bool, int, DataGridFilteringMode, int, T, int, Action<T>, int, Expression<Func<T>>)' cannot be inferred from the usage. Try specifying the type arguments explicitly.`

**Comment by SergeySeleznyov** [comment]
Hi @emysa341,

I've tried to reproduce the issue on the following page:

```razor
@page "/issue62"

<DxComboBox Data="@(new string[] {"Option 1", "Option 2", "Option 3" })" @bind-SelectedItem="SelectedStartStation" NullText="Choose start station" AllowUserInput="true" FilteringMode="DataGridFilteringMode.Contains" ListRenderMode="ListRenderMode.Virtual" @bind-DropDownVisible="@DropDownVisible"></DxComboBox>

@code {
    public string SelectedStartStation { get; set; }
    public bool DropDownVisible { get; set; }
}
```

I seem to skip something important because this page operates properly.

Your can take a look at our troubleshooting about the ```"No generic method 'Take'..."``` message:
[https://github.com/DevExpress/Blazor#3-dxdatagrid-in-blazor](https://github.com/DevExpress/Blazor)

The DxComboBox<T> component uses an IEnumerable<T>﻿ or IQueryable<T>﻿ data source.
Usually, the ```"The type arguments for method ... cannot be inferred from the usage."``` message means that Blazor cannot instantiate some generic class.
The ```"Try specifying the type arguments explicitly."``` recommendation may help in this case, try something like follows:

```razor
<DxComboBox T="@YourDataItemType" ... />
```

If these recommendations do not help, please modify the page about so it reproduces this issue.

Also, to understand the issue deeply I would like to clarify the following details about your project:

* version of DX component NuGet package
* version on .NET Core
* Blazor hosting model (client-side or server-side)

**Comment by emysa341** [comment]
Thanks, <DxComboBox T="@YourDataItemType" ... />, this fixes my issue. Can you post docs link where I can read more about generics and the syntax to use or define that. thanks

**Comment by SergeySeleznyov** [comment]
You are welcome, Ammad.

You can read about generic-typed components here:
[https://docs.microsoft.com/ru-ru/aspnet/core/blazor/components?view=aspnetcore-3.0#generic-typed-components](https://docs.microsoft.com/ru-ru/aspnet/core/blazor/components?view=aspnetcore-3.0)

You can find the DxComboBox documentations here:
[https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxComboBox-1](https://docs.devexpress.com/Blazor/DevExpress.Blazor.DxComboBox-1)

**Comment by emysa341** [comment]
_DevExpress.Blazor v=19.1.8
Asp.NetCore v=3.1.0 LatestPreview
Blazor hosting model (client-side)_

Thanks. I am using following combobox. The "data" contains around 15000 items and when I type something the application hangs for few seconds and then start working normally

  <DxComboBox T="MyType"
                Data="data"
                @bind-SelectedItem="SelectedItem"
                TextFieldName="Name"
                NullText="Choose start station"
                AllowUserInput="true"
                ListRenderMode="ListRenderMode.Virtual"
                FilteringMode="DataGridFilteringMode.Contains">
    </DxComboBox>

Here is console output
`Debugging hotkey: Shift+Alt+D (when application has focus)
blazor.webassembly.js:1 WASM: ﻿Fetching tariiff points.
mono.js:1 [Violation] 'setTimeout' handler took 54ms
blazor.webassembly.js:1 WASM: 15540 tariiff points  fetched.
blazor.webassembly.js:1 WASM: GC_MAJOR_SWEEP: major size: 11104K in use: 18131K
blazor.webassembly.js:1 WASM: GC_MAJOR: (LOS overflow) time 24.60ms, stw 24.64ms los size: 1024K in use: 93K
blazor.webassembly.js:1 WASM: GC_MINOR: (LOS overflow) time 0.50ms, stw 0.55ms promoted 0K major size: 11104K in use: 5086K los size: 1024K in use: 93K
[Violation] 'click' handler took 394ms
esm-chunk-1266fe9a.js:1 [Violation] 'setTimeout' handler took 245ms
esm-chunk-1266fe9a.js:1 [Violation] 'setTimeout' handler took 278ms
17[Violation] 'keyup' handler took <N>ms
blazor.webassembly.js:1 WASM: GC_MINOR: (Nursery full) time 2.35ms, stw 2.38ms promoted 406K major size: 11760K in use: 5922K los size: 1024K in use: 266K
esm-chunk-1266fe9a.js:1 [Violation] 'setTimeout' handler took 668ms
blazor.webassembly.js:1 WASM: GC_MAJOR_SWEEP: major size: 11488K in use: 103049K
blazor.webassembly.js:1 WASM: GC_MAJOR: (LOS overflow) time 33.44ms, stw 33.53ms los size: 1024K in use: 275K
blazor.webassembly.js:1 WASM: GC_MINOR: (Nursery full) time 2.19ms, stw 2.28ms promoted 25K major size: 26720K in use: 23074K los size: 1024K in use: 307K`

**Comment by emysa341** [comment]
At the moment, I am completely unsure which hosting model I am going to use, I am still deciding. That would be great, if there is possibility of binding to a remote source.  When its rolling out ?

**Comment by SergeySeleznyov** [comment]
I've added a backlog note to inform you when this feature is released.

If you do not mind, I will mark this issue as closed since it seems to be solved.
Feel free to contact us if you need further assistance.

**Comment by zabelin-vladimir** [comment]
Hello @emysa341 

> That would be great, if there is possibility of binding to a remote source. 

I'm happy to announce that we implemented the [CustomData](https://docs.devexpress.com/Blazor/DevExpress.Blazor.Base.DxComboBoxBase-1.CustomData) property for DxComboBox in our [latest update (19.2 Beta)](https://community.devexpress.com/blogs/aspnet/archive/2020/01/17/blazor-components-localization-asynchronous-data-operations-and-more-v19-2-0.aspx). 

Use the _CustomData_ property to bind the DxComboBox to a Web API service. Check the [Remote Data Source](https://demos.devexpress.com/Blazor/GridRemoteDataSource#CustomDataSource) demo which illustrates how to use this property with our DataGrid and DxDataGridComboBoxColumn column.

I will be glad to hear your feedback about these events and other features which we implemented.
